### PR TITLE
Mob 4109 create library modules

### PR DESCRIPTION
**Pull request by BartlomiejSopalaKarhoo**
## JIRA
[MOB-4109](https://karhoo.atlassian.net/browse/MOB-4109)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [X ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        .package(name: "KarhooSDK", url: "https://github.com/karhoo/karhoo-ios-sdk", .exact(Version(1, 6, 1))),
+        .package(name: "Adyen", url: "https://github.com/Adyen/adyen-ios", .exact(Version(4, 7, 1))),
+        .package(name: "FloatingPanel", url: "https://github.com/scenee/FloatingPanel", .exact(Version(2, 0, 1))),
+        .package(name: "BraintreeDropIn", url: "https://github.com/braintree/braintree-ios-drop-in", .exact(Version(9, 3, 0))),
+        .package(name: "PhoneNumberKit", url: "https://github.com/marmelroy/PhoneNumberKit", .exact(Version(3, 3, 0))),
+        .package(name: "Braintree", url: "https://github.com/braintree/braintree_ios", .exact(Version(5, 5, 0)))
+    ],
+    targets: [
+        .target(
+            name: "KarhooUISDK",
```
Comment (dianakarh) [inline Package.swift:36]: Could we have targets that already include core + adyen and core + braintree respectively?

**Review by dianakarh — APPROVED** [review]

File: Client/AppDelegate.swift [hunk 2]
```diff
@@ -7,8 +7,12 @@
 
 import KarhooUISDK
 import KarhooSDK
+#if canImport(Braintree)
 import Braintree
+#endif
+#if canImport(Braintree)
```
Comment (dianakarh) [inline Client/AppDelegate.swift:13]: If can import Braintree => Import Adyen 😅

File: Client/AppDelegate.swift [hunk 3]
```diff
@@ -33,10 +39,13 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         if url.scheme?.localizedCaseInsensitiveCompare(urlScheme) == .orderedSame {
             return BTAppSwitch.handleOpen(url, options: options)
         }
+        
+        #if canImport(Braintree)
```
Comment (dianakarh) [inline Client/AppDelegate.swift:43]: Adyen related code in a canImport Braintree block again

File: AdyenPSP/AdyenThreeDSecureUtils.swift [hunk 4]
```diff
@@ -1,11 +1,12 @@
+////
+////  AdyenThreeDSecureUtils.swift
```
Comment (dianakarh) [inline AdyenPSP/AdyenThreeDSecureUtils.swift:2]: Double commented

File: BraintreePSP/BraintreePaymentScreensBuilder.swift [hunk 5]
```diff
@@ -19,21 +20,34 @@ import BraintreePaymentFlow
...
+
             if error != nil {
                 paymentMethodAdded?(.failed(error: error as? KarhooError))
-                //TODO: Check, why it is removed
-           /* } else if result?.isCancelled == true {
-                paymentMethodAdded?(.cancelled(byUser: true))*/
+            } else if isResultCanceled == true {
+                paymentMethodAdded?(.cancelled(byUser: true))
             } else {
                 let nonce = Nonce(nonce: result!.paymentMethod!.nonce, cardType: result!.paymentMethod!.type, lastFour: String(result!.paymentDescription.suffix(2)))
```
Comment (dianakarh) [inline BraintreePSP/BraintreePaymentScreensBuilder.swift:52]: Would prefer an `if let` or `??` on the result and payment method

**Review by AleksanderWedrychowskiKarhoo — DISMISSED** [review]

**Review by AleksanderWedrychowskiKarhoo — APPROVED** [review]

**Review by dianakarh — DISMISSED** [review]

**Review by dianakarh — APPROVED** [review]

**Review by AleksanderWedrychowskiKarhoo — APPROVED** [review]

**Pull request merged**